pinot-query-planner/src/main/java/org/apache/calcite/rel/rules/PinotQueryRuleSets.java

```diff
@@ -115,4 +116,12 @@ private PinotQueryRuleSets() {
           PinotAggregateExchangeNodeInsertRule.INSTANCE,
           PinotWindowExchangeNodeInsertRule.INSTANCE
       );
+
+  public static final Collection<RelOptRule> FILTER_PUSHDOWN_RULES = ImmutableList.of(
```

Comment:
todo: by default Calcite will use depth first order for running these rules. Also it won't do a "fullRestartAfterTransformation" unless we use HepMatchOrder.TOP_DOWN or HepMatchOrder.BOTTOM_UP.

I think using depth first order without doing full restarts after transformation should be fine but would be good if someone else also chimes in. Note that the match order can be changed for only this collection (it's a HepInstruction) so it doesn't need to be a global setting.

Reply:
I think for now we should be good. HEP planner is used here to avoid a lengthy volcano planner that adds latency to the planning phase. as long as the plan results are determinisitc we can always change the way we configure the planner IMO